internal/blocksync/peer_store.go

```diff
@@ -100,24 +89,21 @@ func (p *InMemPeerStore) MaxHeight() int64 {
 	return p.maxHeight
 }
 
-// PeerUpdate applies update functions to the peer if it exists
-func (p *InMemPeerStore) PeerUpdate(peerID types.NodeID, updates ...PeerUpdateFunc) {
-	p.mtx.Lock()
-	defer p.mtx.Unlock()
-	peer, found := p.get(peerID)
+// Update applies update functions to the peer if it exists
+func (p *InMemPeerStore) Update(peerID types.NodeID, updates ...store.UpdateFunc[types.NodeID, PeerData]) {
+	p.store.Update(peerID, updates...)
```

Comment:
looks like inconsistent locking - in `GetAndDelete` and `Delete` we lock before doing anything on p.store.

Reply:
`Store` has its own exclusive mutex. Thus, InMemPeerStore 's the mutex locks only those places where we interract withInMemPeerStore's state